crates/connector_proxy/src/interceptors/airbyte_capture_interceptor.rs

```diff
@@ -1,32 +1,438 @@
-use crate::apis::{
-    FlowCaptureOperation, Interceptor, InterceptorStream, RequestResponseConverterPair,
+use crate::apis::{FlowCaptureOperation, Interceptor, InterceptorStream};
+
+use crate::errors::{create_custom_error, Error};
+use crate::libs::airbyte_catalog::{
+    self, ConfiguredCatalog, ConfiguredStream, Range, ResourceSpec,
+};
+use crate::libs::command::resume_process;
+use crate::libs::json::create_root_schema;
+use crate::libs::protobuf::{decode_message, encode_message};
+use crate::libs::stream::stream_all_airbyte_messages;
+
+use async_stream::stream;
+use bytes::Bytes;
+use protocol::capture::{
+    discover_response, validate_response, DiscoverRequest, DiscoverResponse, Documents,
+    PullRequest, PullResponse, SpecResponse, ValidateRequest, ValidateResponse,
 };
-use crate::errors::Error;
+use protocol::flow::{DriverCheckpoint, Slice};
+use std::collections::HashMap;
+use std::sync::Arc;
+// TODO: switch to std::sync::Mutex;
+use tokio::sync::Mutex;
+
+use futures_util::StreamExt;
+use serde_json::value::RawValue;
+use std::fs::File;
+use std::io::Write;
+use tempfile::{Builder, TempDir};
+use tokio_util::io::StreamReader;
 
-// A placeholder for real logic of airbyte connectors. Details might change during real-implementations.
-pub struct AirbyteCaptureInterceptor {}
+const CONFIG_FILE_NAME: &str = "config.json";
+const CATALOG_FILE_NAME: &str = "catalog.json";
+const STATE_FILE_NAME: &str = "state.json";
+
+pub struct AirbyteCaptureInterceptor {
+    validate_request: Arc<Mutex<Option<ValidateRequest>>>,
+    stream_to_binding: Arc<Mutex<HashMap<String, usize>>>,
+    tmp_dir: TempDir,
+}
 
 impl AirbyteCaptureInterceptor {
-    fn convert_request(
-        _operation: &FlowCaptureOperation,
-        _in_stream: InterceptorStream,
-    ) -> Result<InterceptorStream, Error> {
-        panic!("TBD AirbyteCaptureInterceptor")
+    pub fn new() -> Self {
+        // TODO: keep tempdir inside container only, not mounted to any local dir.
+        // -- tmpfs mount? https://docs.docker.com/storage/tmpfs/
```

Comment:
just came across `tmpfs` that stores files in host memory being another way of storing secure infomation in docker. Shall we look more into that?